packages/dds/tree/src/simple-tree/leafNodeSchema.ts

```diff
@@ -80,6 +80,19 @@ function makeLeaf<Name extends string, const T extends ValueSchema>(
 	return new LeafNodeSchema(`com.fluidframework.leaf.${name}`, t);
 }
 
+export function isLeafNodeSchema(
+	schema: TreeNodeSchema,
+): schema is LeafNodeSchema<string, ValueSchema> {
+	return schema.kind === NodeKind.Leaf;
+}
+
+/**
+ * Returns the given schema as a {@link LeafNodeSchema} and throws an error if it isn't one.
+ */
+export function asLeafNodeSchema(schema: TreeNodeSchema): LeafNodeSchema<string, ValueSchema> {
+	return isLeafNodeSchema(schema) ? schema : fail("Schema is not a leaf schema");
```

Comment:
if you do keep this, I think factoring it like this is cleaner:

```suggestion
	assert(isLeafNodeSchema(schema), "Schema is not a leaf schema);
	return schema;
```

That said, I think it make more sense to not have this utility and for the caller to just do the assert themselves instead of using this. It avoids having to declare an extra variable with the different type, and better matches the conventions used for other schema kinds.